clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1449,7 +1449,8 @@ protected void updateOffsetsFromConsumerRecord(
       if (!isDaVinciClient) {
         // also update the leader topic position using the upstream position in LeaderMetadata
         PubSubPosition newUpstreamPosition = extractUpstreamPosition(consumerRecord);
-        if (!PubSubSymbolicPosition.EARLIEST.equals(newUpstreamPosition)) {
+        if (!PubSubSymbolicPosition.EARLIEST.equals(newUpstreamPosition)
+            && extractUpstreamClusterId(consumerRecord) != -1) {
```

Comment:
What is the context of this?  Just that the clusterId doesn't exist because there is no upstream?